markets/models.py

```diff
@@ -101,6 +115,9 @@ def has_started(self) -> bool:
     def has_ended(self) -> bool:
         return timezone.now() >= self.end_date
 
+    @property
+    def is_upcoming(self) -> bool:
+        return timezone.now() < self.start_date and timezone.now()+timedelta(days=7) >= self.start_date
```

Comment:
Let's contract this to `timezone.now() < self.start_date < timezone.now() + timedelta(days=7)`, easier to read and shorter